### PR TITLE
CIRC-7014 - Add Missing SMR Transactions

### PR DESCRIPTION
**Pull request by pjulien**
Add SMR transactions to Lua API

### CODE REVIEW

File: src/modules/lua_check.c [hunk 1]
```diff
@@ -205,6 +207,7 @@ noit_lua_module_set_xml_description(lua_State *L) {
   else if(lua_gettop(L) > 1)
     luaL_error(L, "wrong number of arguments");
```
Comment (postwait) [inline src/modules/lua_check.c:208]: This will longjmp out without ending the SMR transaction.

File: src/modules/lua_check.c [hunk 2]
```diff
@@ -430,6 +443,8 @@ noit_lua_set_metric_f(lua_State *L, mtev_boolean allow_whence,
   int64_t __l = 0;
   uint64_t __L = 0;
 
+  mtev_memory_begin();
+
   if(lua_gettop(L) < 2 || lua_gettop(L) > 4) luaL_error(L, "need 2-4 arguments: <metric_name> <value> [whence_s] [whence_us]");
```
Comment (postwait) [inline src/modules/lua_check.c:448]: We can't error inside an SMR transaction b/c it will leave without closing it.

**Review by postwait — CHANGES_REQUESTED** [review]

File: src/modules/lua_check.c (hunk 2 above)
Comment (pjulien) [reply src/modules/lua_check.c:448]: I thought this would kill the process

Comment (postwait) [reply src/modules/lua_check.c:448]: No, it will break the runtime just for this check execution.  the top level lua_resume will return an error.

Comment (pjulien) [reply src/modules/lua_check.c]: Yes, I see, it does a ``_longjmp``

**Review by pamaddox — APPROVED** [review]

**Pull request merged**